src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo, User } from './types/types';
+import { TodoList } from './components/TodoList';
+
+const getMaxId = (todos: Todo[]) => {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+};
+
+const isWhitespaceString = (str: string) => !str.replace(/\s/g, '').length;
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [users] = useState<User[]>(usersFromServer);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [formData, setFormData] = useState({
+    id: 0,
+    title: '',
+    completed: false,
+    userId: 0,
+  });
+
+  const reset = () => {
+    setFormData({
+      id: 0,
+      title: '',
+      completed: false,
+      userId: 0,
+    });
+
+    setHasUserIdError(false);
+    setHasTitleError(false);
+  };
+
+  const handleAdd = ({ id, ...allData }: Todo) => {
+    const newTodo = {
+      id: getMaxId(todos),
+      ...allData,
+      userId: +allData.userId,
+    };
+
+    setTodos(currentTodo => [...currentTodo, newTodo]);
+  };
+
+  const handleSetInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    setHasTitleError(!value || isWhitespaceString(value));
+    setFormData(currentUser => ({ ...currentUser, [name]: value }));
+  };
+
+  const handleSetSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = e.target;
+
+    setHasUserIdError(+value === 0);
+    setFormData(currentUser => ({ ...currentUser, [name]: value }));
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!formData.title || isWhitespaceString(formData.title));
+    setHasUserIdError(+formData.userId === 0);
+
+    if (
+      !formData.title ||
```

Comment:
```suggestion
      !formData.title.trim() ||
```
<img width="252" alt="image" src="https://github.com/mate-academy/react_add-todo-form/assets/70220667/517330b0-1eaa-4517-9ce6-fdf9eb8ae104">
to fix empty todo